src/MagnumPlugins/AssimpImporter/Test/AssimpImporterTest.cpp

```diff
@@ -287,6 +349,690 @@ void AssimpImporterTest::openDataFailed() {
+        "animation.gltf")));
+
+    std::ostringstream out;
+    {
+        Warning redirectWarning{&out};
+        CORRADE_VERIFY(importer->animation(1));
+    }
+    CORRADE_VERIFY(out.str().find(" ticks per second is incorrect for glTF, patching to 1000\n")
+        != std::string::npos);
+}
+
+void AssimpImporterTest::animationGltfBrokenSplineWarning() {
+    if(!supportsAnimation(".gltf"))
+        CORRADE_SKIP("glTF 2 animation is not supported with the current version of Assimp");
+
+    if(!ASSIMP_HAS_BROKEN_GLTF_SPLINES)
+        CORRADE_SKIP("Current version of assimp correctly imports glTF spline-interpolated animations.");
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("AssimpImporter");
+
+    std::ostringstream out;
+    {
+        Warning redirectWarning{&out};
+        CORRADE_VERIFY(importer->openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR,
+            "animation.gltf")));
+    }
+    CORRADE_COMPARE(out.str(), "Trade::AssimpImporter::openData(): spline-interpolated animations imported "
+        "from this file are most likely broken using this version of Assimp. Consult the "
+        "importer documentation for more information.\n");
+}
+
+void AssimpImporterTest::animationGltfSpline() {
+    if(!supportsAnimation(".gltf"))
+        CORRADE_SKIP("glTF 2 animation is not supported with the current version of Assimp");
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("AssimpImporter");
+    CORRADE_VERIFY(importer->openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR,
+        "animation.gltf")));
+
+    CORRADE_COMPARE(importer->animationCount(), 3);    
+    CORRADE_COMPARE(importer->animationName(2), "TRS animation, splines");
+
+    constexpr Float keys[]{ 0.5f, 3.5f, 4.0f, 5.0f };
+
+    auto animation = importer->animation(2);
+    CORRADE_VERIFY(animation);
+    CORRADE_VERIFY(animation->importerState());
+    /* Four T/R/S keys */
+    CORRADE_COMPARE(animation->data().size(),
+        4*(sizeof(Float) + sizeof(Quaternion)) +
+        2*4*(sizeof(Float) + sizeof(Vector3)));
+    CORRADE_COMPARE(animation->trackCount(), 3);
+
+    /* Rotation */
+    CORRADE_COMPARE(animation->trackType(0), AnimationTrackType::Quaternion);
+    CORRADE_COMPARE(animation->trackResultType(0), AnimationTrackType::Quaternion);
+    CORRADE_COMPARE(animation->trackTargetType(0), AnimationTrackTargetType::Rotation3D);
+    CORRADE_COMPARE(animation->trackTarget(0), 3);
+    Animation::TrackView<const Float, const Quaternion> rotation = animation->track<Quaternion>(0);
+    CORRADE_COMPARE(rotation.interpolation(), Animation::Interpolation::Linear);
+    CORRADE_COMPARE(rotation.before(), Animation::Extrapolation::Constant);
+    CORRADE_COMPARE(rotation.after(), Animation::Extrapolation::Constant);
+    CORRADE_COMPARE_AS(rotation.keys(), (Containers::StridedArrayView1D<const Float>{keys}), TestSuite::Compare::Container);
+    if(!ASSIMP_HAS_BROKEN_GLTF_SPLINES) {
```

Comment:
```suggestion
    {
      #if ASSIMP_HAS_BROKEN_GLTF_SPLINES
      CORRADE_EXPECT_FAIL("Current version of assimp incorrectly imports glTF spline-interpolated animations.");
      #endif
```

Again, to show that this is a known importer failure in the test output :) The scope is important as it limits what comparisons the expected failure applies to.

Same below.

Reply:
Done